crypto/s2n_openssl.c

```diff
@@ -0,0 +1,42 @@
+/*
```

Comment:
Should this file be named `s2n_libcrypto.c`?

Reply:
It's the source file for methods in s2n_openssl.h, but maybe they should both be "libcrypto" instead of "openssl". Although the defines we're using _are_ all OPENSSL_IS_ ;)

Reply:
Or do you think the new methods should be called s2n_openssl_is_ instead?

Reply:
I agree, it's weird this file is called s2n_openssl.c

Reply:
I renamed s2n_openssl.c, but I didn't rename s2n_openssl.h. I can do a follow-up PR to rename s2n_openssl.h if we think it's necessary, but I don't want to include that change in this PR because it touches a LOT of files.